File: NotificationSystem/NotificationSystem.Web/Controls_Add/ctrAgent_Add.ascx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Data;
using NotificationSystem.NotificationSystem.Data.NotificationSystemTableAdapters;
using static NotificationSystem.NotificationSystem.Data.NotificationSystem;
using NotificationSystem.NotificationSystem.Data.Classes;
using System.Diagnostics;
using System.Globalization;
using System.IO;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Security;
using System.Text;
using System.Threading.Tasks;
using Microsoft.VisualBasic;
using Telerik.Web.UI;
using System.Data.SqlClient;
using System.Net.Http;


    public partial class ctrAgent_Add : System.Web.UI.UserControl
    {

        private SqlDataReader dr;

        private int m_AgentID = 0;

        public int AgentID
        {
            get
            {
                return m_AgentID;
            }
            set
            {
                m_AgentID = value;
            }
        }


        public TextBox txtAgentList { get; set; }
	public int AgentList { get; internal set; }

	protected void Page_Load(object sender, EventArgs e)
        {
            clsNotificationSystem theNotificationSystem = new clsNotificationSystem();
            AgentDataTable tblAgent = new AgentDataTable();
            var client = new HttpClient();
            try
            {
                if ((Page.IsPostBack))
                {
                    //button click to add form data ?

                    if (Request.Form["ctl00$MainContent$ctrAgent_Add$btnAdd"] == "Add")
                        AddAgent();
                }
                else
                {
                PopulateControls();
                }
            }
            catch (Exception ex)
            {
                throw;
            }
        }

        public void AddAgent()
        {
            try
            {
                clsAgent thisAgent = new clsAgent();

                {
                    var withBlock = thisAgent;
                    if (txtagentid.Text.Length == 0)
                        return;


                    withBlock.AgentID = int.Parse(txtagentid.Text); withBlock.Title = txttitle.Text; withBlock.FirstN = txtfirstname.Text; withBlock.LastN = txtlastname.Text; withBlock.Email = txtemailaddress.Text; withBlock.Phone = txtPhoneNumber.Text; withBlock.Fax = txtFaxNumber.Text;
                    withBlock.Salary = int.Parse(txtsalary.Text); 
                }

                try
                {
                    clsNotificationSystem theNotificationSystem = new clsNotificationSystem();
                    theNotificationSystem.AddAgent(thisAgent);
                    lblResult.Text = "Agent data has been added";
                }
                catch (SqlException ex)
                {
                    if (ex.Number == 2627)
                        lblResult.Text = "Agent already exist!";
                    else
                        throw new ApplicationException(ex.Message);
                }
            }


            catch (Exception ex)
            {
                clsNotificationSystem_Web SendError = new clsNotificationSystem_Web();
                string NotificationBody = ex.Message + "  " + ex.StackTrace;
                SendError.SendMailMessage(NotificationBody);
                Response.Redirect("ErrorPage.aspx", false);
            }
        }
    public void PopulateControls()
	{
        clsNotificationSystem theNotificationSystem = new clsNotificationSystem();
    }
        public void ClearControls()
        {
            try
            {

            }


            catch (Exception ex)
            {
                throw;
            }
        }

        protected void btnCancel_Click(object sender, EventArgs e)
        {
            Response.Redirect("Agent_Find.aspx", false);
        }
    }

File: NotificationSystem/NotificationSystem.Web/Controls_Add/ctrAgent_Add.ascx.vb
Imports System
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports NotificationSystem.NotificationSystem.Data.NotificationSystem
Imports NotificationSystem.NotificationSystem.Data.Classes
Imports System.Data.SqlClient

Public Partial Class ctrAgent_Add
    Inherits UserControl
    Private _AgentList As Integer



    Private m_AgentID As Integer = 0

    Public Property AgentID As Integer
        Get
            Return m_AgentID
        End Get
        Set(value As Integer)
            m_AgentID = value
        End Set
    End Property


    Public Property txtAgentList As TextBox

    Public Property AgentList As Integer
        Get
            Return _AgentList
        End Get
        Friend Set(value As Integer)
            _AgentList = value
        End Set
    End Property

    Protected Sub Page_Load(sender As Object, e As EventArgs)
        Dim theNotificationSystem As clsNotificationSystem = New clsNotificationSystem()
        Dim tblAgent As AgentDataTable = New AgentDataTable()

        Try
            If MyBase.Page.IsPostBack Then
                If Equals(Request.Form("ctl00$MainContent$ctrAgent_Add$btnAdd"), "Add") Then AddAgent()
            Else
                PopulateControls()
            End If
        Catch __unusedException1__ As Exception
            Throw
        End Try
    End Sub

    Public Sub AddAgent()
        Try
            Dim thisAgent As clsAgent = New clsAgent()

            If True Then
                Dim withBlock = thisAgent
                If txtagentid.Text.Length = 0 Then Return


                withBlock.AgentID = Integer.Parse(txtagentid.Text)
                withBlock.Title = txttitle.Text
                withBlock.FirstN = txtfirstname.Text
                withBlock.LastN = txtlastname.Text
                withBlock.Email = txtemailaddress.Text
                withBlock.Phone = txtPhoneNumber.Text
                withBlock.Fax = txtFaxNumber.Text
                withBlock.Salary = txtsalary.Text
            End If

            Try
                Dim theNotificationSystem As clsNotificationSystem = New clsNotificationSystem()
                theNotificationSystem.AddAgent(thisAgent)
                lblResult.Text = "Agent data has been added"
            Catch ex As SqlException
                If ex.Number = 2627 Then
                    lblResult.Text = "Agent already exist!"
                Else
                    Throw New ApplicationException(ex.Message)
                End If
            End Try


        Catch ex As Exception
            Dim SendError As clsNotificationSystem_Web = New clsNotificationSystem_Web()
            Dim NotificationBody = ex.Message & "  " & ex.StackTrace
            SendError.SendMailMessage(NotificationBody)
            Response.Redirect("ErrorPage.aspx", False)
        End Try
    End Sub
    Public Sub PopulateControls()
        Dim theNotificationSystem As clsNotificationSystem = New clsNotificationSystem()
    End Sub
    Public Sub ClearControls()
        Try


        Catch __unusedException1__ As Exception
            Throw
        End Try
    End Sub

    Protected Sub btnCancel_Click(sender As Object, e As EventArgs)
        Response.Redirect("Agent_Find.aspx", False)
    End Sub
End Class
